content/300-guides/050-database/100-developing-with-prisma-migrate/175-squashing-migrations.mdx

```diff
@@ -0,0 +1,124 @@
+
+### Migrating cleanly from a development environment
+
+Squashing migrations can be useful when developing with a branch-based workflow. During a large local development effort on a feature branch you might generate multiple migrations using `migrate dev`. After the feature is finished, the migration history might contain unnecessary intermediate steps that are unwanted in the final migration history that will be pushed to the `main` branch.
+
+There could be important reasons to avoid applying the intermediate steps in production — they might lose data or be extremely slow / disruptive). Even when this is not the case, you may want to avoid clutter in your production environment's migrations history.
+
+For detailed steps on how to achieve this using `migrate diff`, see the section on [how to migrate cleanly from a development environment](#how-to-migrate-cleanly-from-a-development-environment).
+
+### Creating a clean history in a production environment
+
+Squashing migrations can also be used in a production environment to squash all migration files into one. This can be useful when the production enviroment has accumulated a longer migration history, and replaying it in new environments has become a burden due to intermediate steps requiring extra time. Since the team is not deriving value from the migration steps (and could get them back from version control history in a pinch) the decision is made to squash the whole history into a single migration.
+
+For detailed steps on how to achieve this using `migrate diff` and `migrate resolve` see the section on [how to create a clean history in a production environment](#how-to-create-a-clean-history-in-a-production-environment).
+
+## Considerations when squashing migrations
+
+<Admonition type="warning">
+
+When squashing migrations, be aware that any custom SQL in your `migration.sql` files will not be retained.
+
+</Admonition>
+
+## How to squash migrations
+
+This section provides step-by-step instructions on how to squash migrations in the two scenarios discussed above:
+
+- [Migrating cleanly from a development environment](#how-to-migrate-cleanly-from-a-development-environment)
+- [Creating a clean history in a production environment](#how-to-create-a-clean-history-in-a-production-environment)
+
+### How to migrate cleanly from a development environment
+
+Before squashing your migrations, make sure you have the following starting conditions:
+
+- The contents of the migrations to be squashed are not yet applied on the production database
+- All migrations applied to production are part of the local migration history already
+- There is no custom SQL in the migration files
+
+Then follow these steps:
+
+1. Reset the contents of your local `./migrations` folder to match the migration history on the `main` branch
+
+2. Create an empty `migration.sql` file in the `./migrations` folder and make a note of the file path
```

Comment:
Would this be done manually currently? And just to check, does this involve making a new folder with the `<timestamp>_name` format, and putting a new empty `migration.sql` file inside of it?

Reply:
Yes, currently there is no command for this that we offer. And you're correct its the folder plus the file.

Reply:
Added in the folder name.